### PR TITLE
test(e2e): add i18n e2e tests

### PR DESCRIPTION
**Pull request by joshuaellis**
### What does it do?

* adds e2e tests for i18n that test the feature as part of the application through various veritcal journeys
* enhances the settings table with better accessibility labels
* if you try to go to a locale that does not exist, we redirect you to the default in the edit & list views
* fixes situations where the locale injected UI was not correctly disabled
* fixes issue with UID field where the default value kept overwrite what ever we wanted the regeneration to be
* fixes issue with form not being populated with defaults when creating a new locale on an existing document
* fixes issue where no temp keys were being added to repeatable components or dynamic zones

### CODE REVIEW

File: packages/core/admin/admin/src/content-manager/pages/EditView/utils/data.ts [hunk 1]
```diff
@@ -117,8 +118,20 @@ const prepareRelations = traverseData(
...
 const prepareTempKeys = traverseData(
-  () => false,
-  (v) => v
+  (attribute) =>
+    (attribute.type === 'component' && attribute.repeatable) || attribute.type === 'dynamiczone',
+  (data) => {
+    if (Array.isArray(data) && data.length > 0) {
+      const keys = generateNKeysBetween(undefined, undefined, data.length);
+
+      return data.map((datum, index) => ({
```
Comment (jhoward1994) [inline packages/core/admin/admin/src/content-manager/pages/EditView/utils/data.ts:127]: I love the word datum

File: packages/core/admin/admin/src/content-manager/pages/EditView/components/FormInputs/DynamicZone/Field.tsx [hunk 2]
```diff
@@ -180,28 +180,34 @@ const DynamicZone = ({
...
-    //         'There {number, plural, =0 {are # missing components} one {is # missing component} other {are # missing components}}',
-    //     },
-    //     { number: missingComponentNumber }
-    //   );
-    // }
+    if (hasError && dynamicDisplayedComponentsLength > max) {
+      return formatMessage({
+        id: getTranslation(`components.DynamicZone.extra-components`),
+        defaultMessage:
+          'There {number, plural, =0 {are # extra components} one {is # extra component} other {are # extra components}}',
```
Comment (jhoward1994) [inline packages/core/admin/admin/src/content-manager/pages/EditView/components/FormInputs/DynamicZone/Field.tsx:197]: Do you need to pass the number param here too?

Comment (joshuaellis) [reply packages/core/admin/admin/src/content-manager/pages/EditView/components/FormInputs/DynamicZone/Field.tsx:197]: yes, i do.

File: e2e/tests/i18n/editview.spec.ts [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+    await page
+      .getByRole('textbox', { name: 'name' })
+      .fill('Camiseta de fuera 23/24 de Nike para hombres');
+
+    /**
+     * Verify the UID works as expected
+     */
+    await expect
+      .poll(async () => {
+        const requestePromise = page.waitForRequest('**/content-manager/uid/generate?locale=es');
```
Comment (jhoward1994) [inline e2e/tests/i18n/editview.spec.ts:66]:
```suggestion
        const requestPromise = page.waitForRequest('**/content-manager/uid/generate?locale=es');
```

Or maybe requeste is spanish? :D

**Comment by jhoward1994** [comment]
Do we want to handle navigating to the default locale even for invalid locale query strings?

e.g. if I modify the URL i can get an error like

Invalid locale format: notALocale

In the CM

**Review by jhoward1994 — APPROVED** [review]
Very nice!

File: e2e/tests/i18n/editview.spec.ts [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+        await page.getByRole('button', { name: 'Regenerate' }).click();
+        const req = await requestePromise;
+        return req.postDataJSON();
+      })
+      .toMatchObject({
+        contentTypeUID: 'api::product.product',
+        data: {
+          id: '',
+          isAvailable: true,
+          name: 'Camiseta de fuera 23/24 de Nike para hombres',
```
Comment (Marc-Roig) [inline e2e/tests/i18n/editview.spec.ts:76]: ✌️

File: e2e/tests/i18n/listview.spec.ts [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+test.describe('List view', () => {
+  test.beforeEach(async ({ page }) => {
+    await resetDatabaseAndImportDataFromPath('./e2e/data/with-admin.tar');
+    await page.goto('/admin');
+    await login({ page });
+  });
+
+  const LOCALES = ['English (en)', 'French (fr)', 'German (de)', 'Spanish (es)'];
+
+  test('As a user I want a locale column to be injected in the table & a locale switcher only if the content-type has i18n enabled.', async ({
```
Comment (Marc-Roig) [inline e2e/tests/i18n/listview.spec.ts:14]: should we test if the available locales column is listed in the table?

**Review by Marc-Roig — APPROVED** [review]
I like the e2e format, they are so easy to read :)

**Comment by joshuaellis** [comment]
I think going forward we'll embellish these tests to assert unhappy paths too, for now this asserts core functionality is there 😄

**Pull request merged**